jupyterlab_git/git.py

```diff
@@ -222,6 +222,10 @@ async def config(self, path, **kwargs):
 
         return response
 
+    def check_ignore(self, path):
+        # in git.py Git, add a new method check_ignore that execute git check-ignore -z <path> and parse the result to return the list of ignored files; changed_files method is a good example of a similar method.
+        return
```

Comment:
What you would like to do is to execute the git `check-ignore` command in the repository to know if it is ignored or not. Something like this should do:

```suggestion
    def check_ignore(self, repository_path, path):
       """
       Check if a path is ignored or not.
       
       Args:
         repository_path: Path to the Git repository
         path: Relative path to check
       """
        # in git.py Git, add a new method check_ignore that execute git check-ignore -z <path> and parse the result to return the list of ignored files; changed_files method is a good example of a similar method.

        cmd = ["git", "check-ignore", "-z", path]

        response = {}
        try:
            code, output, error = await execute(cmd, cwd=repository_path)
        except subprocess.CalledProcessError as e:
            response["code"] = e.returncode
            response["message"] = e.output.decode("utf-8")
        else:
            response["code"] = code
            if code != 0:
                response["command"] = " ".join(cmd)
                response["message"] = error
            else:
                response["files"] = strip_and_split(output)

        return response
```